common/graphics/texture.cc

```diff
@@ -61,11 +77,14 @@ void TextureRegistry::OnGrContextDestroyed() {
     it.second->OnGrContextDestroyed();
   }
 
-  for (const auto& [id, weak_image] : images_) {
+  for (const auto& [id, pair] : ordered_images_) {
+    auto index_id = pair.first;
+    auto weak_image = pair.second;
     if (auto image = weak_image.lock()) {
       image->OnGrContextDestroyed();
     } else {
-      images_.erase(id);
+      image_indices_.erase(index_id);
+      ordered_images_.erase(id);
```

Comment:
Here and elsewhere, mutation while enumeration makes me nervous. Are iterators on `ordered_images_` invalidated when the erase happens here?

Reply:
Used to doing this safely by using iterators explicitly and assigning the iterator to the return call of `erase`. I am not sure I have ever done it this way. So want to double check this is safe.

Reply:
Yeah, I'll change this. It seems to work but it'd be safer to use the iterator.

I had another issue with mutation during iteration that I fixed by copying the map into a vector, but that would be overkill here.